zerver/tests/test_users.py

```diff
@@ -1703,6 +1703,26 @@ def test_get_user_profile(self) -> None:
         self.assertEqual(result['user']['email'], bot.email)
         self.assertTrue(result['user']['is_bot'])
 
+    def test_get_user_by_email(self) -> None:
+        user = self.example_user('hamlet')
+        self.login('hamlet')
+        result = orjson.loads(self.client_get(f'/json/users/{user.delivery_email}').content)
+        self.assertEqual(result['user']['email'], user.email)
+
+        self.assertEqual(result['user']['full_name'], user.full_name)
+        self.assertIn("user_id", result['user'])
+        self.assertNotIn("profile_data", result['user'])
+        self.assertFalse(result['user']['is_bot'])
+        self.assertFalse(result['user']['is_admin'])
+        self.assertFalse(result['user']['is_owner'])
+
+        result = orjson.loads(self.client_get(f"/json/users/{user.delivery_email}", {"include_custom_profile_fields": "true"}).content)
```

Comment:
Shouldn't we verify that the `custom_profile_fields` were included here?

I see the previous test has the same bug, maybe do a prep commit improving that test, and then add this?

Reply:
I have added <code>self.assertIn('profile_data', result['user'])</code> to verify that <code>custom_profile_fields</code> are included. I find that this is already present in the previous test. Do we need to test anything else as well.